curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-color-markers-set/step-050.md

```diff
@@ -0,0 +1,87 @@
+---
+id: 61a493ead935c148d2b76312
+title: Step 50
+challengeType: 0
+dashedName: step-50
+---
+
+# --description--
+
+The `linear-gradient` function is very flexible -- here is the basic syntax you'll use in this tutorial:
+
+```css
+linear-gradient(gradientDirection, color1, color2, ...);
+```
+
+`gradientDirection` is the direction of the line used for the transition. `color1` and `color2` are color arguments, which are the colors that will be used in the transition itself. These can be any type of color, including color keywords, hex, `rgb`, or `hsl`.
+
+Now you'll apply a red-to-green gradient along a 90 degree line to the first marker.
+
+First, in the `.red` CSS rule, set the `background` property to `linear-gradient()`, and pass it the value `90deg` as the `gradientDirection`.
+
+# --hints--
+
+Your `.red` CSS rule should have a `background` property with the value `linear-gradient(90deg)`.
+
+```js
+assert.equal(__helpers.CSSHelp(document).getStyle('.red').getPropVal('background', true), 'linear-gradient(90deg)');
+```
```

Comment:
This test has actually pointed out that this is not valid CSS - a `linear-gradient` with just an angle

What should we do?

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient()

Reply:
We may need to regex this, then.

Reply:
We can do, but are we fine having the lesson end with incorrect CSS?

Reply:
I think there's a lot to teach with linear gradient, and it would be too much for a single step. So yes.